### PR TITLE
Feature/gh157 dshost l3 c test code generation

### PR DESCRIPTION
**Pull request by hgfell683**
dshost l3 c test code generation & Spec

### CODE REVIEW

File: src/test_l3_dsHost.c [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+   // Step 1: Call dsGetCPUTemperature()
+   status = dsGetCPUTemperature(&cpuTemperature);
+   UT_LOG_INFO("Result dsGetCPUTemperature: dsError_t:[%s]",
+                 UT_Control_GetMapString(dsError_mapTable, status));
+    UT_LOG_INFO("CPU Temperature: %f\n", cpuTemperature);
+
+    UT_ASSERT_EQUAL_FATAL(status, dsERR_NONE);
+
+   if(prevTemp == 0)
```
Comment (hari22yuva) [inline src/test_l3_dsHost.c:169]: in L3 verification need to implement in Python

File: docs/pages/ds-host_L3_Low-Level_TestSpec.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+- `HAL Interface Header File` - [dsHost.h](https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsHost.h)
+
+## Level 3 Test Cases High Level Overview
+
+Each test case need to verify with the each supported video port.
+Below are top test use-case for the video port.
+
+|#|Test-case|Description|HAL APIs|Source|Sink|
+|-|---------|-----------|--------|------|----|
+|1|Verify that the get temperature function |After the device is running for sometime, call getTemperature, and call it after a five minute wait. The two temperatures should not see any noticable changes in temperature |`dsGetCPUTemperature()`|`Y`|`Y`|
```
Comment (hari22yuva) [inline docs/pages/ds-host_L3_Low-Level_TestSpec.md:35]: No need to check for this module required L3 test?

File: src/main.c [hunk 3]
```diff
@@ -100,6 +101,13 @@ int main(int argc, char** argv)
         return -1;
     }
 
+    registerReturn = UT_register_APIDEF_l3_tests();
```
Comment (hari22yuva) [inline src/main.c:104]: please re-base branch, registration is already done

File: src/test_l3_dsHost.c [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    UT_add_test( pSuite, "L3_Init_dsHost", test_l3_dsHost_hal_Init);
+    UT_add_test( pSuite, "L3_GetTemperature", test_l3_dsHost_hal_get_Temperature);
+    UT_add_test( pSuite, "L3_GetSocID", test_l3_dsHost_hal_get_SocID);
+    UT_add_test( pSuite, "L3_GetHostEdid", test_l3_dsHost_hal_get_hostEdid);
+    UT_add_test( pSuite, "L3_Term_dsHost", test_l3_dsHost_hal_Term);
+
+    return 0;
+}
+
+/** @} */ // End of DS_Host_HALTEST_L2
```
Comment (hari22yuva) [inline src/test_l3_dsHost.c:321]: End of DS_Host_HALTEST_L3

File: src/test_l3_dsHost.c [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+*
+*/
+void test_l3_dsHost_hal_Init(void)
+{
+   gTestID = 1;
+   dsError_t status = dsERR_NONE ;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+   // Step 1: Call dsHostInit()
```
Comment (hari22yuva) [inline src/test_l3_dsHost.c:125]: each function has calling & result except callbacks 
https://github.com/rdkcentral/rdk-halif-test-device_settings/blob/develop/src/test_l3_dsVideoPort.c#L324

File: src/test_l3_dsHost.c [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+ * @{
+ */
+
+/**
+ * @defgroup DS_Host_HALTEST Device Settings Host HAL Tests
+ * @{
+ */
+
+/**
+ * @defgroup DS_Host_HALTEST_L2 Device Settings Host HAL Tests L2 File
```
Comment (hari22yuva) [inline src/test_l3_dsHost.c:50]: update the defgroup 
please remove L2 instances in the file

**Review by hari22yuva — COMMENTED** [review]
Please check the review comments

File: docs/pages/ds-host_L3_Low-Level_TestSpec.md (hunk 2 above)
Comment (hgfell683) [reply docs/pages/ds-host_L3_Low-Level_TestSpec.md:35]: What do you mean? And yes, this is the L3 test we have noted for dsHost

File: src/test_l3_dsHost.c (hunk 1 above)
Comment (hgfell683) [reply src/test_l3_dsHost.c:171]: What do you mean?

File: src/main.c (hunk 3 above)
Comment (hgfell683) [reply src/main.c]: Rebased and updated.

File: src/test_l3_dsHost.c (hunk 5 above)
Comment (hgfell683) [reply src/test_l3_dsHost.c:125]: Updated with calling logs.

File: src/test_l3_dsHost.c (hunk 6 above)
Comment (hgfell683) [reply src/test_l3_dsHost.c]: Updated as requested

File: src/test_l3_dsHost.c [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+void test_l3_dsHost_hal_get_Temperature(void)
```
Comment (akhilbhas) [inline src/test_l3_dsHost.c:155]: Will this step called twice in L3 test ?

Can you point me to spec so I can understand test ?

Comment (hgfell683) [reply src/test_l3_dsHost.c:155]: @akhilbhas Yes, this is to be called more then once within the L3 test suite. It's to gauge to make sure the tempature of the device remains stable when in the same mode of use for an extended period of time. See:

|1|Verify that the get temperature function |After the device is running for sometime, call getTemperature, and call it after a five minute wait. The two temperatures should not see any noticable changes in temperature |`dsGetCPUTemperature()`|`Y`|`Y`|

It's also the only L3 test we have specified for dsHost. I know there was some discussion how necessary it was/is

File: src/test_l3_dsHost.c (hunk 1 above)
Comment (bhanucbp) [reply src/test_l3_dsHost.c]: test_l3_dsHost_hal_get_Temperature(C code) function just prints the current temperature, user/python calls  test_l3_dsHost_hal_get_Temperature multiple times(using the menu) and decides whether the temperature raise is within the accepted limit or not

Comment (hgfell683) [reply src/test_l3_dsHost.c]: Updated as requested

File: docs/pages/dsHost/ds-host_L3_Low-Level_TestSpec.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  - For more details refer [RAFT](https://github.com/rdkcentral/python_raft/blob/1.0.0/README.md) and [example_device_config.yml](https://github.com/rdkcentral/python_raft/blob/1.0.0/examples/configs/example_device_config.yml)
+
+- **componentProfile.yaml/platformProfile.yaml**
+  - Contains component-specific configurations
+  - Contains platform wide configuration broken down into separate components
+  - Example configuration file [dsHost_Settings](https://github.com/rdkcentral/rdk-halif-test-device_settings/blob/3.0.0/profiles/sink/Sink_AudioSettings.yaml)
+
+- **testSetupConfig.yaml**
+  - This configuration file contains the list of requirements for tests to execute. Eg: Copying the streams, setting environment variables etc.
+  - Example configuration file [dsHost_L3_testSetup.yml](../../../host/tests/dsHost_L3_Tests/dsHost_L3_testSetup.yml)
```
Comment (hari22yuva) [inline docs/pages/dsHost/ds-host_L3_Low-Level_TestSpec.md:92]: Update the files

File: docs/pages/dsHost/ds-host_L3_Low-Level_TestSpec.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+- **testSuite.yaml**
+  - This configuration file contains the list of menu items for C/C++ L3 test running on `DUT`
+  - Example configuration file [dsHost_test_suite.yml](../../../host/tests/dsClasses/dsHost_test_suite.yml)
```
Comment (hari22yuva) [inline docs/pages/dsHost/ds-host_L3_Low-Level_TestSpec.md:96]: Update the files

File: src/test_l3_dsHost.c [hunk 10]
```diff
@@ -0,0 +1,315 @@
...
+    gTestID = 3;
+    dsError_t status = dsERR_NONE ;
+
+    char* socID[DSHOST_SOC_LENGTH] = {0};
+
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    // Step 1: Call dsGetSoCId()
+    UT_LOG_INFO("Calling dsGetSoCId(IN:socID[])");
+    status = dsGetSoCId(socID, DSHOST_SOC_LENGTH);
```
Comment (hari22yuva) [inline src/test_l3_dsHost.c:203]: dsGetSoCId() is available ? in host

**Review by hari22yuva — COMMENTED** [review]
please address the review comments

File: docs/pages/dsHost/ds-host_L3_Low-Level_TestSpec.md (hunk 8 above)
Comment (hgfell683) [reply docs/pages/dsHost/ds-host_L3_Low-Level_TestSpec.md:87]: Created and did initial update for files

File: docs/pages/dsHost/ds-host_L3_Low-Level_TestSpec.md (hunk 9 above)
Comment (hgfell683) [reply docs/pages/dsHost/ds-host_L3_Low-Level_TestSpec.md:91]: Updated

File: host/tests/dsHost_L3_Tests/dsHost_L3_testSetup.yml [hunk 11]
```diff
@@ -0,0 +1,20 @@
+# this file is generated by the testing suites, when a new test is modified, this configuration file will be filled out and updated.
+#location: https://github.com/rdkcentral/rdk-halif-test-device_settings/tree/3.1.2/assets/
+
+dsCompositeIn:  # Prefix must always exist
+  description: "dsCompositeIn Device Settings test setup"
```
Comment (hari22yuva) [inline host/tests/dsHost_L3_Tests/dsHost_L3_testSetup.yml:5]: please keep dshost module test setup file

Comment (hgfell683) [reply host/tests/dsHost_L3_Tests/dsHost_L3_testSetup.yml]: Updated

**Review by hari22yuva — DISMISSED** [review]
ship it

File: docs/pages/dsHost/ds-host_L3_Low-Level_TestSpec.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+  - Specifies overall configuration for the platform
+  - Can be overridden by:
+    - Changing locally .yaml file directory
+    - Using --deviceConfig command line switch
+  - For more details refer [RAFT](https://github.com/rdkcentral/python_raft/blob/1.0.0/README.md) and [example_device_config.yml](https://github.com/rdkcentral/python_raft/blob/1.0.0/examples/configs/example_device_config.yml)
+
+- **testSetupConfig.yaml**
+  - This configuration file contains the list of requirements for tests to execute. Eg: Copying the streams, setting environment variables etc.
+  - Example configuration file [dsHost_L3_testSetup.yml](../../../host/tests/dsHost_L3_Tests/dsHost_L3_testSetup.yml)
+
```
Comment (hari22yuva) [inline docs/pages/dsHost/ds-host_L3_Low-Level_TestSpec.md:88]: componentProfile.yaml/platformProfile.yaml
section is missing

Comment (hgfell683) [reply docs/pages/dsHost/ds-host_L3_Low-Level_TestSpec.md:93]: Should be updated.

**Review by hari22yuva — APPROVED** [review]
ship it

**Pull request merged**